Roadmap content (#9241)
ethereum/ethereum-org-website: src/content/roadmap/index.md

lang: en
template: roadmap
image: ../../assets/upgrades/merge.png
alt: "Ethereum roadmap"
summaryPoints:
buttons:
  - label: Further updates
    toId: what-changes-are-coming
  - label: Past updates
    to: /history/
    variant: outline
---

Ethereum is already a powerful platform for global coordination, but it is still being improved. An ambitious set of improvements will upgrade Ethereum from its current form into a fully scaled, maximally resilient platform. These upgrades are laid out in the Ethereum roadmap.

**To learn about previous upgrades to Ethereum, please visit our [History](/history/) page**

## What changes are coming to Ethereum? {#what-changes-are-coming}

The Ethereum roadmap outlines the specific improvements that will be made to protocol in the future. Overall, the roadmap will bring the following benefits to Ethereum users:

<CardGrid>
  <RoadmapActionCard
    to="/roadmap/scaling"
    title="Cheaper transactions"
    image="scaling"
    description="Rollups are too expensive and rely on centralized components, causing users to place too much trust in their operators. The roadmap includes fixes for both of these problems."
  >

>     <ButtonLink to="/roadmap/scaling/">Read more</ButtonLink>
>   </RoadmapActionCard>
>   <RoadmapActionCard
>     to="/roadmap/security"
>     title="Extra security"
>     image="security"
>     description="Ethereum is already very secure but it can be made even stronger, ready to withstand all kinds of attack far into the future."
>   >
>     <ButtonLink to="/roadmap/security/">Read more</ButtonLink>
>   </RoadmapActionCard>
>   <RoadmapActionCard
>     to="/roadmap/user-experience"
>     title="Better user experience"
>     image="userExperience"
>     description="More support for smart contract wallets and light-weight nodes will make using Ethereum simpler and safer."
>   >
>     <ButtonLink to="/roadmap/user-experience/">Read more</ButtonLink>
>   </RoadmapActionCard>
>   <RoadmapActionCard
>     to="/roadmap/future-proofing"
>     title="Future proofing"
>     image="futureProofing"
>     description="Ethereum researchers and developers are solving tomorrow's problems today, readying the network for future generations."
>   >
>     <ButtonLink to="/roadmap/future-proofing/">Read more</ButtonLink>

Reviewer: ```suggestion
    <ButtonLink to="/roadmap/scaling/">More on reducing fees</ButtonLink>
  </RoadmapActionCard>
  <RoadmapActionCard
    to="/roadmap/security"
    title="Extra security"
    image="security"
    description="Ethereum is already very secure but it can be made even stronger, ready to withstand all kinds of attack far into the future."
  >
    <ButtonLink to="/roadmap/security/">More on security</ButtonLink>
  </RoadmapActionCard>
  <RoadmapActionCard
    to="/roadmap/user-experience"
    title="Better user experience"
    image="userExperience"
    description="More support for smart contract wallets and light-weight nodes will make using Ethereum simpler and safer."
  >
    <ButtonLink to="/roadmap/user-experience/">More on user experience</ButtonLink>
  </RoadmapActionCard>
  <RoadmapActionCard
    to="/roadmap/future-proofing"
    title="Future proofing"
    image="futureProofing"
    description="Ethereum researchers and developers are solving tomorrow's problems today, readying the network for future generations."
  >
    <ButtonLink to="/roadmap/future-proofing/">More on future proofing</ButtonLink>
```

<RoadmapImageContent title="How the roadmap is defined">

The roadmap is mostly the result of years of work by researchers and developers - because the protocol is very technical - but any motivated person can participate. Ideas usually start off as discussions on a forum such as [ethresear.ch](https://github.com/ethereum/ethereum-org-website/blob/1c5502a0abaa746512323bbce0fbf1093387a724/src/content/roadmap/ethresear.ch), [Ethereum magicians](https://www.figma.com/exit?url=https%3A%2F%2Fethereum-magicians.org%2F) or the Eth R&D discord server. They may be responses to new vulnerabilities that are discovered, suggestions from organizations working in the application layer (such as dapps and exchanges) or from known frictions for end users (such as costs or transaction speeds). When these ideas mature, they can be proposed as [Ethereum Improvement Proposals](https://eips.ethereum.org/). This is all done in public so that anyone from the community can weigh in at any time.

Read more on [Ethereum governance](https://ethereum.org/en/governance/)

</RoadmapImageContent>

> <InfoBanner>

Reviewer 2: We're missing a gap here when this component is rendered before a header

```suggestion
<InfoBanner mb={8}>
```

One way to think about Ethereum development is by analogy to biological evolution. A network that is able to adapt to new challenges and maintain fitness is more likely to succeed that one that is resistant to change, although as the network becomes more and more performant, scalable and secure fewer changes to the protocol will be required.

## Do I have to do anything when there is an upgrade?

Upgrades tend not to impact end-users except by providing better user-experiences and a more secure protocol and perhaps more <i>options</i> for how to interact with Ethereum. End uses are not required to actively participate in an upgrade, nor are they required to do anything to secure their assets. Node operators will need to update their clients to prepare for an upgrade. Some upgrades may lead to changes for application developers. For example, history expiry upgrades may lead application developers to grab historical data from new sources.

## What about The Verge, The Splurge, etc?

> Vitalik Buterin proposed a vision for the Ethereum roadmap that was organized into several categories linked by their effects on Ethereum's architecture. It included:

Reviewer: ```suggestion
[Vitalik Buterin proposed a vision for the Ethereum roadmap](https://twitter.com/VitalikButerin/status/1588669782471368704) that was organized into several categories linked by their effects on Ethereum's architecture. It included:
```

---
title: Ethereum roadmap
description: The path to more scalability, security and sustainability for Ethereum.
lang: en
template: roadmap
image: ../../assets/upgrades/merge.png
alt: "Ethereum roadmap"
summaryPoints:
  - Learn about how Ethereum will evolve into the future
  - Read why Ethereum is still being worked on

>   - Understand how Ethereum is setting up for long term success

Reviewer: Same comment as on account abstraction. We should be summarising, not telling users the objectives of the page.

cc: @konopkja

Author: ah - yea this is placeholder frontmatter that should be deleted, the bullets are not actually used on the page.

buttons:
  - label: Further updates
    toId: what-changes-are-coming
  - label: Past updates
    to: /history/
    variant: outline
---

> Ethereum is a powerful platform for global coordination, but it is not finished! An ambitious set of improvements will upgrade Ethereum from its current form into a fully scaled, maximally resilient platform. These upgrades are laid out in the Ethereum roadmap.

Reviewer: Positive language 

```suggestion
Ethereum is already a powerful platform for global coordination, but it is still being improved. An ambitious set of improvements will upgrade Ethereum from its current form into a fully scaled, maximally resilient platform. These upgrades are laid out in the Ethereum roadmap.
```

> **To learn about previous upgrades to Ethereum, such as EIP-1559 or The Merge, please visit our [History](/history/) page**

Reviewer: Is this accurate if we're not transitioning pages to the History page in this PR?

## What changes are coming to Ethereum? {#what-changes-are-coming}

The Ethereum roadmap outlines the specific improvements that will be made to protocol in the future. Overall, the roadmap will bring the following benefits to Ethereum users:

> <CardGrid>

Reviewer: Overall, I think the content of these cards should be pithier (or just generally a little shorter).

  <RoadmapActionCard
    to="/roadmap/scaling"

>     title="Faster, cheaper transactions"

Reviewer: @jmcook1186 @konopkja 

Feel free to ignore if you disagree, but I'd personally focus on one benefit here as the hook. The major issue people are concerned with isn't the speed with which a transaction goes through, it's how much it costs them in gas fees.

>     title="Faster, cheaper transactions"

Reviewer: ```suggestion
    title="Cheaper transactions"
```

    image="scaling"

>     description="Today's scaling solutions, known as rollups, are too expensive because of the way they send data to Ethereum. They also rely on some centralized components meaning users have to place too much trust in small groups of operators. The roadmap includes fixes for both of these problems."

Reviewer: ```suggestion
    description="Rollups are too expensive and rely on centralized components, causing users to place too much trust in their operators. The roadmap includes fixes for both of these problems."
```

  >
    <ButtonLink to="/roadmap/security/">Read more</ButtonLink>
  </RoadmapActionCard>
  <RoadmapActionCard
    to="/roadmap/user-experience"
    title="Improved user experience"
    image="userExperience"
    description="Using Ethereum will soon be as intuitive as the rest of the web. Ethereum nodes will be able to run on small devices like mobile phones, giving every user trustless access to their data and assets. Changes to the way users use accounts will also make using Ethereum simpler and safer."
  >
    <ButtonLink to="/roadmap/user-experience/">Read more</ButtonLink>
  </RoadmapActionCard>
  <RoadmapActionCard
    to="/roadmap/future-proofing"
    title="Future proofing"
    image="futureProofing"
    description="Today's Ethereum researchers and developers are building a protocol for future generations. This means anticipating problems that could arise in decades to centuries and solving them now, such as quantum computers that break current crypotography."
  >
    <ButtonLink to="/roadmap/future-proofing/">Read more</ButtonLink>
  </RoadmapActionCard>
</CardGrid>

> ## Why Ethereum needs a roadmap {#why-ethereum-needs-a-roadmap}

Reviewer: Feels a bit rhetorical. But fine with keeping it...

> Ethereum is a network that undergoes periodic upgrades that enhance its scalability, security or sustainability. One of the core strengths of Ethereum is that it is not fixed forever in its current form - as new ideas arise from research and development they can be implemented to make sure Ethereum is ever-improving. This gives Ethereum the flexibility to tackle emerging challenges and keep up with the most advanced technological breakthroughs in cryptography, computing and cryptoeconomics.

Reviewer: ```suggestion
Ethereum gets regular upgrades that enhance its scalability, security, or sustainability. One of Ethereum's core strengths is adapting as new ideas emerge from research and development. Adaptability gives Ethereum the flexibility to tackle emerging challenges and keep up with the most advanced technological breakthroughs.
```

When these ideas mature, they can be proposed as [Ethereum Improvement Proposals](https://eips.ethereum.org/). Client developers then create prototypes that get investigated intensely and simulated on private and then public testnets before being scheduled for releasing on Ethereum Mainnet. There are nine independent Ethereum client teams (four execution clients and five consensus clients) that all do this independently in different languages. This is all done in public so that anyone from the community can weigh in at any time.

</RoadmapImageContent>

<InfoBanner>
  <h4 style="margin-top: 0">What is ETH2?</h4>

  <p>The term 'Eth2' was commonly used to describe the future of Ethereum before the switch to proof-of-stake but it was <strong>phased out in favor of more precise terminology.</strong> It was originally used to differentiate the Ethereum network before the switch to proof-of-stake and the network after, or sometimes to refer to the different Ethereum clients (execution clients were sometimes referred to as ETH1 clients and consensus clients were sometimes referred to as ETH2 clients).</p>

>   <p style="margin-bottom: 0">Some staking operators have also represented ETH staked on the Beacon Chain with the ‘ETH2’ ticker. This creates potential confusion, given that users of these services are not actually receiving an ‘ETH2’ token. No ‘ETH2’ token exists; it simply represents their share in that specific providers’ stake.</p>

Reviewer: Unnecessary?

Author: yes

</InfoBanner>

## Is the roadmap set in stone?

> No! The roadmap is the current plan for upgrading Ethereum in the near term and also far into the future. The roadmap can change as new information and new technology becomes available. For example, the original Ethereum roadmap intended for sharding to be implemented before The Merge. However, scaling by rollups developed much faster than expected and the roadmap was updated to focus on layer 2 scaling first, allowing The Merge to happen sooner than originally anticipated. The roadmap is a set of intentions for improving Ethereum - it is the research and developer community's best guess about where we should go from here.

Reviewer: Avoid metaphors when unnecessary: _Set in stone_ -> '**Is the roadmap fixed**?', or, to use more precise and positive language: '**Can the roadmap change**?'

Author: ```suggestion
## Is the roadmap fixed?

No! The roadmap is the current plan for upgrading Ethereum in the near term and also far into the future. The roadmap can change as new information and new technology becomes available. For example, the original Ethereum roadmap intended for sharding to be implemented before The Merge. However, scaling by rollups developed much faster than expected and the roadmap was updated to focus on layer 2 scaling first, allowing The Merge to happen sooner than originally anticipated. The roadmap is a set of intentions for improving Ethereum - it is the research and developer community's best guess about where we should go from here.
```

> No! The roadmap is the current plan for upgrading Ethereum in the near term and also far into the future. The roadmap can change as new information and new technology becomes available. For example, the original Ethereum roadmap intended for sharding to be implemented before The Merge. However, scaling by rollups developed much faster than expected and the roadmap was updated to focus on layer 2 scaling first, allowing The Merge to happen sooner than originally anticipated. The roadmap is a set of intentions for improving Ethereum - it is the research and developer community's best guess about where we should go from here.

Reviewer: > For example, the original Ethereum roadmap intended for sharding to be implemented before The Merge. However, scaling by rollups developed much faster than expected, and core developers updated the roadmap to focus on layer 2 scaling first, allowing The Merge to happen sooner than originally anticipated. 

I don't think this context is necessary here.

> ## Is the roadmap set in stone?
> 
> No! The roadmap is the current plan for upgrading Ethereum in the near term and also far into the future. The roadmap can change as new information and new technology becomes available. For example, the original Ethereum roadmap intended for sharding to be implemented before The Merge. However, scaling by rollups developed much faster than expected and the roadmap was updated to focus on layer 2 scaling first, allowing The Merge to happen sooner than originally anticipated. The roadmap is a set of intentions for improving Ethereum - it is the research and developer community's best guess about where we should go from here.

Reviewer: Reworded based on the above.

```suggestion
## Will Ethereum's roadmap change over time?

Yes—almost definitely. The roadmap is the current plan for upgrading Ethereum, covering both near-term and future plans. We expect the roadmap to change as new information and technology become available. 

Think of Ethereum's roadmap as a set of intentions for improving Ethereum; it is the core researchers' and developers' best hypothesize of Ethereum's most optimal path forward.
```

## When will the roadmap be finished?

> The current roadmap contains some upgrades that will be implemented in the next 6 months (e.g. staking withdrawals) and others that will probably take 5-10 years to be fully implemented (e.g. quantum resistance). The precise timing of each upgrade is very difficult to predict as many roadmap items are worked on in parallel and develop at different speeds. The relative urgency of different upgrades can also change over time depending on external factors (e.g. a sudden leap in the performance and availability of quantum computers may make quantum resistant cryptography more urgent).

Reviewer: ```suggestion
Ethereum will implement some upgrades in the next six months (e.g. staking withdrawals); others are lower priority and likely not to be implemented for the next 5-10 years (e.g. quantum resistance). Giving precise timing of each upgrade is complicated to predict as many roadmap items are worked on in parallel and developed at different speeds. The urgency of an upgrade can also change over time depending on external factors (e.g. a sudden leap in the performance and availability of quantum computers may make quantum-resistant cryptography more urgent).
```

Reviewer: > (e.g. a sudden leap in the performance and availability of quantum computers may make quantum resistant cryptography more urgent).

Not sure this analogy will really land with most people

Reviewer: Not blocking, but maybe using scaling as an example would be better here

One way to think about Ethereum development is by analogy to biological evolution. A network that is able to adapt to new challenges and maintain fitness is more likely to succeed that one that is resistant to change, although as the network becomes more and more performant, scalable and secure fewer changes to the protocol will be required.

## Do I have to do anything when there is an upgrade?

Upgrades tend not to impact end-users except by providing better user-experiences and a more secure protocol and perhaps more <i>options</i> for how to interact with Ethereum. End uses are not required to actively participate in an upgrade, nor are they required to do anything to secure their assets. Node operators will need to update their clients to prepare for an upgrade. Some upgrades may lead to changes for application developers. For example, history expiry upgrades may lead application developers to grab historical data from new sources.

> ## What about the verge, the splurge etc?

Reviewer: Good thinking to include a reference to this here.

```suggestion
## What about Vitalik Buterin's roadmap?
```

```suggestion
## What about The Verge, The Splurge, etc?
```

> Vitalik proposed a vision for the Ethereum roadmap that was organized into several categories linked by their effects on Ethereum's artchitecture. It included:

Reviewer: ```suggestion
Vitalik Buterin proposed a vision for the Ethereum roadmap that was organized into several categories linked by their effects on Ethereum's architecture. It included:
```

- The Merge: upgrades relating to the switch from proof-of-work to proof-of-stake
- The Surge: upgrades related to scalability by rollups and data sharding
- The Scourge: upgrades related to censorship resistance, decentralization and protocol risks from MEV
- The Verge: upgrades related to verifying blocks more easily
- The Purge: upgrades related to reducing the computational costs of running nodes and simplifying the protocol

> - The Splurge: other upgrades that don't fit well into the previous categories.

Reviewer: @jmcook1186 an afterthought (not sure if we do it already), but for v2 we should plan to include this terminology on the appropriate pages if it's relevant (i.e. 'x' aka 'the verge') for discoverability

> We decided not to persist this terminology on this page because we wanted to use a simpler and more user-centric model.

Reviewer: ```suggestion
We decided not to use this terminology because we wanted to use a simpler and more user-centric model. Although we use user-centric language, the vision remains the same as the one proposed by Vitalik.
```

**To learn about previous upgrades to Ethereum, such as EIP-1559 or The Merge, please visit our [History](/history/) page**

## What changes are coming to Ethereum? {#what-changes-are-coming}

The Ethereum roadmap outlines the specific improvements that will be made to protocol in the future. Overall, the roadmap will bring the following benefits to Ethereum users:

<CardGrid>
  <Card title="Faster, cheaper transactions" emoji="🐟">

    Today's rollups are still in their "training wheels" phase where critical components are run by centralized operators. Decentralizing the rollup infrastructure is a critical part of scaling Ethereum without compromising on Ethereum's core ethos of decentralization. At the same time, it is too expensive for rollups to post their data to Ethereum and this limits how cheaply users can transact. Adding data-blobs to blocks and sharding data across nodes will solve this problem, leading to near-instant, <$0.001 transactions for users.

>     [Read how Ethereum will scale](/roadmap/faster-transactions/)

Reviewer 3: Would it make sense to call this page `/roadmap/scaling/` vs. `/roadmap/faster-transactions/`?

Reviewer 3: Haven't looked but I suspect search volume around "Ethereum scaling" is higher than terms around faster transactions.

Author: done!

Some staking operators have also represented ETH staked on the Beacon Chain with the ‘ETH2’ ticker. This creates potential confusion, given that users of these services are not actually receiving an ‘ETH2’ token. No ‘ETH2’ token exists; it simply represents their share in that specific providers’ stake.
</ExpandableCard>

<ExpandableCard title="What about the verge, the splurge etc?">
Vitalik proposed a vision for the Ethereum roadmap that was organized into several categories linked by their effects on Ethereum's artchitecture. It included:

- The Merge: upgrades relating to the switch from proof-of-work to proof-of-stake
- The Surge: upgrades related to scalability by rollups and data sharding
- The Scourge: upgrades related to censorship resistance, decentralization and protocol risks from MEV
- The Verge: uprades related to verifyign blocks more easily
- The Purge: upgrades related to reducing the computational costs of running nodes and simplifying the protocol
- The Splurge: other upgrades that don't fit well into the previous categories.

> We decided not to persist this terminology on this page because we wanted to use a simpler and more user-centric model.

Reviewer: ❤️
